staging/src/k8s.io/apiserver/pkg/endpoints/handlers/workflow.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"net/http"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// guessWorkflowID tries to find a reasonable default for the
+// workflowID, based on user-agent of the incoming request,
+// sub-resource, verb and content-type. Always returns a non-empty
+// string.
```

Comment:
Additional things we have to worry about:
* Whoops, I got the default workflow ID, but now I want to deliberately manage my workflow--how do I switch it over?
* Accumulation of cruft-- when & how to truncate abandoned workflows?

Maybe don't belong as part of this, but definitely in a doc or the changelog or something.

Reply:
I don't want to go in a long discussion about this orthogonal subject here :-) 
But yeah, this might be a problem, I can add a todo.